Model/Schedule.php

```diff
@@ -143,33 +150,36 @@ public function nextOpening($fileRate, $seekDays, $startDay = 0)
                 $start->modify($timeFrom);
 
-                // Evaluate if we have exceeded allowed file count for this day.
-                $fileCount = $this->evaluateFileRate($fileRate);
-
-                // If we have already built a file in this day and can send more...
-                if ($fileCount > 0 && $fileRate > 1) {
-                    // Push the start time to the next available slot in this day.
-                    $daySeconds = $end->format('U') - $start->format('U');
-                    if ('00:00' === $timeFrom && '23:59' === $timeTill) {
-                        // Avoid sending 2 files at midnight.
-                        $segmentSeconds = intval($daySeconds / $fileRate);
-                    } else {
-                        // Send at opening and closing times, spreading the rest of the day evenly.
-                        $segmentSeconds = intval($daySeconds / ($fileRate - 1));
+                if ('file' === $this->contactClient->getType()) {
+                    // Evaluate if we have exceeded allowed file count for this day.
+                    $fileCount = $this->evaluateFileRate($fileRate, $date);
+
+                    // If we have already built a file in this day and can send more...
+                    if ($fileCount > 0 && $fileRate > 1) {
+                        // Push the start time to the next available slot in this day.
+                        $daySeconds = $end->format('U') - $start->format('U');
+                        if ('00:00' === $timeFrom && '23:59' === $timeTill) {
+                            // Avoid sending 2 files at midnight.
+                            $segmentSeconds = intval($daySeconds / $fileRate);
+                        } else {
+                            // Send at opening and closing times, spreading the rest of the day evenly.
+                            $segmentSeconds = intval($daySeconds / ($fileRate - 1));
+                        }
+                        // Push start time to the next segment.
+                        $start->modify('+'.($segmentSeconds * $fileCount).' seconds');
                     }
-                    // Push start time to the next segment.
-                    $start->modify('+'.($segmentSeconds * $fileCount).' seconds');
                 }
 
                 // Start time should not be in the past.
-                if (0 === $day && $start < $realNow) {
+                if (0 === $day && $start < $this->now) {
                     // Must be done after rate has been applied.
-                    $start = $realNow;
+                    $start = $this->now;
                 }
 
-                // Return the next appropriate window to send contacts.
-                return [$start, $end];
-                break;
+                $openings[] = [$start, $end];
+                if (!$all) {
+                    break;
+                }
             } catch (\Exception $e) {
                 if ($e instanceof ContactClientException) {
```

Comment:
You can have multiple catches for a try catch.
`try { // .... }

catch (\Exception $e) { // ... }

catch (ContactClientException) { // ...  }`